Modules/Logger/include/selxLogger.h

```diff
@@ -57,34 +58,43 @@ class Logger : public itk::DataObject
   /** Run-time type information (and related methods). */
   itkTypeMacro( Self, itk::DataObject );
 
-  /* m_Logger is initialized in the default constructor */
+  /* m_LoggerImpl is initialized in the default constructor */
   Logger();
-  LoggerImpl & GetLogger( void );
 
   /** The actual logger is a pimpled member variable */
-  typedef std::shared_ptr< LoggerImpl > LoggerPointer;
-
-  typedef const std::string ChannelType;
-  typedef const std::string FormatType;
-  typedef const std::string MessageType;
-
-  void AddConsole( FormatType format = "[%LineID% %TimeStamp%]: %Message%" );
-
-  // void AddFile( FileNameType fileName = "SuperElastix_%Y-%m-%d_%H-%M-%S.%N.log",
-  //               FormatType format = "[%LineID% %TimeStamp% %ComponentName% %Channel% %SeverityLevel%]: %Message%" );
-  // void AddFile( FileNameType fileName = "SuperElastix_%Y-%m-%d_%H-%M-%S.%N.log",
-  //               LoggerImpl::ChannelType channel = "SuperElastix",
-  //               FormatType format = "[%LineID% %TimeStamp% %ComponentName% %Channel% %SeverityLevel%]: %Message%" );
-
-  virtual void Log( SeverityType severity, MessageType message );
-
-  // void Log( ChannelType channel, SeverityType severity, MessageType message );
+  typedef std::unique_ptr< LoggerImpl > LoggerImplPointer;
 
+  /* Provide a reference to internal logger */
+  LoggerImpl & GetLogger( void );
 
+  void SetLogLevel( const LogLevel& level );
```

Comment:
Logger should inherit from user-side base class and component-side, then we can expose only some function to components. Fx components should not be able to change loglevel.